crates/matrix-sdk/src/event_cache/redecryptor.rs

```diff
@@ -403,6 +415,13 @@ impl EventCache {
             }
         }
 
+        let event_ids: BTreeSet<_> =
```

Comment:
Same.

Reply:
I think this was about the early return as well, if so, fixed in https://github.com/matrix-org/matrix-rust-sdk/pull/5746/commits/c3fb01d35d44d77485bcf0227019ddb4978d0887 as well.